### PR TITLE
Tagaccvm_updates

### PR DESCRIPTION
**Pull request by desjardi**
This pull request has both the addition of a clear function and getMoments

### CODE REVIEW

File: irl/c_interface/generic_cutting/c_generic_cutting.cpp [hunk 1]
```diff
@@ -442,6 +442,22 @@ void c_getNormMoments_CapDod_LocSepLink_TagAccVM_SepVM(
           IRL::C_CUTTING_METHOD);
 }
 
+void c_getMoments_CapDod_LocSepLink_TagAccVM_SepVM(
+    const c_CapDod* a_capped_dodecahedron,
+    const c_LocSepLink* a_localized_separator_link,
+    c_TagAccVM_SepVM* a_moments_to_return) {
+  assert(a_capped_dodecahedron != nullptr);
+  assert(a_capped_dodecahedron->obj_ptr != nullptr);
+  assert(a_localized_separator_link != nullptr);
```
Comment (robert-chiodi) [inline irl/c_interface/generic_cutting/c_generic_cutting.cpp:451]: Can you also add a `assert(a_localized_separator_link->obj_ptr != nullptr);`?

File: irl/fortran_interface/f_getvolumemoments.f90 [hunk 2]
```diff
@@ -684,6 +686,18 @@ subroutine F_getNormMoments_CapDod_LocSepLink_TagAccVM_SepVM(a_Capped_Dod, a_loc
     end subroutine F_getNormMoments_CapDod_LocSepLink_TagAccVM_SepVM
   end interface
 
+  interface
+    subroutine F_getMoments_CapDod_LocSepLink_TagAccVM_SepVM(a_Capped_Dod, a_localized_separator_link, a_moments_to_return) &
+    bind(C, name="c_getMoments_CapDod_LocSepLink_TagAccVM_SepVM")
+      use, intrinsic :: iso_c_binding
+      import
+      implicit none
+      type(c_CapDod) :: a_Capped_Dod ! Pointer to CapDod object
```
Comment (robert-chiodi) [inline irl/fortran_interface/f_getvolumemoments.f90:695]: I understand this is how it is in other places (which I need to fix), but could you make the variable name lowercase, `a_capped_dod`?

File: irl/fortran_interface/f_getvolumemoments.f90 [hunk 3]
```diff
@@ -2051,6 +2065,18 @@ subroutine getNormMoments_CapDod_LocSepLink_TagAccVM_SepVM(a_Capped_Dod, a_local
 
   end subroutine getNormMoments_CapDod_LocSepLink_TagAccVM_SepVM
 
+  subroutine getMoments_CapDod_LocSepLink_TagAccVM_SepVM(a_Capped_Dod, a_localized_separator_link, a_moments_to_return)
+   use, intrinsic :: iso_c_binding
+   implicit none
+     type(CapDod_type), intent(in) :: a_Capped_Dod
```
Comment (robert-chiodi) [inline irl/fortran_interface/f_getvolumemoments.f90:2071]: Same comment, `a_capped_dod`.

**Review by robert-chiodi — COMMENTED** [review]
Thanks for the added functionality! Just a couple of quick comments.

File: irl/c_interface/generic_cutting/c_generic_cutting.cpp (hunk 1 above)
Comment (desjardi) [reply irl/c_interface/generic_cutting/c_generic_cutting.cpp:451]: done

File: irl/fortran_interface/f_getvolumemoments.f90 (hunk 3 above)
Comment (desjardi) [reply irl/fortran_interface/f_getvolumemoments.f90:2071]: done.

File: irl/fortran_interface/f_getvolumemoments.f90 (hunk 2 above)
Comment (desjardi) [reply irl/fortran_interface/f_getvolumemoments.f90:695]: done.

**Review by robert-chiodi — APPROVED** [review]
Thanks!

**Pull request merged**